src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -448,10 +448,13 @@ public async Task<Either<ActionResult, Unit>> DeleteUser(string email)
                         .Where(invite => invite.Email.ToLower() == email.ToLower())
                         .ToListAsync();
 
-                    // Delete the internal EES user, if found.
+                    // We don't remove the ContentDb's Users entry. All other stuff related to the
+                    // user is removed. This is done so if that user's Id exists in other table's
+                    // Created/Updated/DeletedById columns, that info isn't lost and we don't have to worry about
+                    // referential integrity (which we would if we removed the User table entry).
                     if (internalUser != null)
                     {
-                        _contentDbContext.Users.Remove(internalUser);
+                        internalUser.Deleted = DateTime.UtcNow;
```

Comment:
Shouldn't we also do a check above to only find an internal user that's not already flagged as deleted, and return an error if no user is found?

At present,
1. This endpoint could be run for an email that's never existed and it will return 204 as though it's been successful.
2. This endpoint could be run for an email which has already deleted and it will 'refresh' the Deleted date with the current time, overwriting the date it was actually deleted, as well as returning 204.

Reply:
I've updated the code so the `SoftDeleted` date cannot be refreshed.

The existing behaviour was that if someone tried to delete a user that didn't exist, it would return a 204, and I went along with that. I assume this was done as I believe we've had cases where the Identity Framework user existed and the app user didn't exist, and where the app user did exist, and the Identity Framework user didn't. In cases where the data got in a mess, we wanted delete user to clean up any instances of a user for us.

I've added a case where we return not found if no identity user *and* no app user are found.

Happy to do something else if you're were thinking of something different.